src/UI/Settings.js

```diff
@@ -1081,7 +1110,11 @@ function Settings(props) {
         </Form>
         <Form onSubmit={handleSubmit}>
           <H3>Change text light</H3>
-          <Input placeholder="Hex or string" ref={textLightInput} />
+          <Input
+            defaultValue={theme.text_light}
```

Comment:
You should not rely on the theme, instead rely on the useTheme() object that has access to all the theme attributes like this:
useTheme().primary_color
You have multiple examples in this file to see how it's used.